.github/workflows/ci.yml

```diff
@@ -89,81 +93,168 @@ jobs:
+          else
+            releasetype=release
+          fi
+          echo RELEASETYPE=$releasetype >> $GITHUB_OUTPUT
+        id: releasetype
 
-      - name: Make Release
+      - name: "Build Debian packages"
         env:
           GOBIN: /usr/local/bin
         run: |
-          make release
+          GOPATH=$PWD/.go go generate -v
+          if [ "${{ steps.releasetype.outputs.RELEASETYPE }}" = "ci" ]; then
+            DEBEMAIL="CI <runner@github>" dch -v `make version` "CI build"
+          fi
+          buildtype="any"
+          if [ "${{ matrix.arch }}" = "amd64" ]; then  # build 'all' type package only on amd64
+            buildtype="any,all"
+          fi
+          dpkg-buildpackage -us -uc --build=$buildtype -d --host-arch=${{ matrix.arch }}
+          mkdir -p build && mv ../*.deb build/
+          cd build && ls -l *.deb
 
-      - name: Publish nightly release to aptly
+      - name: "Publish CI release to aptly"
         if: github.ref == 'refs/heads/master'
         env:
           APTLY_USER: ${{ secrets.APTLY_USER }}
           APTLY_PASSWORD: ${{ secrets.APTLY_PASSWORD }}
         run: |
-          ./upload-artifacts.sh nightly ${{ matrix.suite }}
+          ./upload-artifacts.sh ci ${{ matrix.suite }}
 
-      - name: Publish release to aptly
-        if: startsWith(github.event.ref, 'refs/tags')
+      - name: "Publish release to aptly"
+        if: steps.releasetype.outputs.RELEASETYPE == 'release' && github.ref == 'refs/heads/master'
         env:
           APTLY_USER: ${{ secrets.APTLY_USER }}
           APTLY_PASSWORD: ${{ secrets.APTLY_PASSWORD }}
         run: |
           ./upload-artifacts.sh release ${{ matrix.suite }}
+
+  ci-binary-build:
+    name: "Build"
+    needs: test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        goos: [linux, freebsd, darwin]
+        goarch: ["386", "amd64", "arm", "arm64"]
+        exclude:
+          - goos: darwin
+            goarch: 386
+          - goos: darwin
+            goarch: arm
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v3
+        with:
+          # fetch the whole repo for `git describe` to work
+          fetch-depth: 0
+
+      - name: "Read go version from go.mod"
+        run: |
+          echo "GOVER=$(sed -n 's/^go \(.*\)/\1/p' go.mod)" >> $GITHUB_OUTPUT
+        id: goversion
+
+      - name: "Get aptly version"
+        run: |
+           echo "VERSION=$(make version)" >> $GITHUB_OUTPUT
+        id: releaseversion
+
+      - name: "Setup Go"
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ steps.goversion.outputs.GOVER }}
+
+      - name: "Build aptly ${{ matrix.goos }}/${{ matrix.goarch }}"
+        env:
+          GOBIN: /usr/local/bin
+        run: |
+          make binaries GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }}
+          cd build; unzip aptly_${{ steps.releaseversion.outputs.VERSION }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
```

Comment:
I mentioned elsewhere, but this is breaking the archives actually uploading in the gh-release step. The double-zip only appears in the github actions output; once you upload the zips to a release, they are formatted correctly.